torchgeo/datasets/glacier_mapping_alps.py

```diff
@@ -0,0 +1,427 @@
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if the ``split``, ``cv_iter``, ``version``, ``bands`` or ``extra_features`` are invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            DependencyNotFoundError: If xarray is not installed.
+        """
+        self.root = Path(root)
+        self.split = split
+        self.cv_iter = cv_iter
+        self.version = version
+        self.bands = bands
+        self.extra_features = extra_features
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        # sanity checks
+        assert split in self.valid_splits, f'Split {split} not in: {self.valid_splits}'
+        assert cv_iter in self.valid_cv_iters, (
+            f'Cross-validation iteration {cv_iter} not in: {self.valid_cv_iters}'
+        )
+        assert version in self.valid_versions, (
+            f'Version {version} not in: {self.valid_versions}'
+        )
+        for band in bands:
+            assert band in self.all_bands, f'Band {band} not in: {self.all_bands}'
+        if extra_features:
+            for feature in extra_features:
+                assert feature in self.valid_extra_features, (
+                    f'Feature {feature} not in: {self.valid_extra_features}'
+                )
+
+        # set the local file paths
+        label = f'dataset_{version}'
+        self.fp_archive = self.root / f'{label}.tar.gz'
+        self.dir_patches = self.root / label
+        self.fp_splits_csv = self.root / 'splits.csv'
+
+        # get the corresponding urls and checksums
+        self.url_dataset = self.download_metadata[label]['url']
+        self.md5_dataset = self.download_metadata[label]['checksum']
+        self.url_csv_splits = self.download_metadata['splits_csv']['url']
+        self.md5_csv_splits = self.download_metadata['splits_csv']['checksum']
+
+        self._verify()
+        self._prepare_files()
+
+    def __len__(self) -> int:
+        """The length of the dataset.
+
+        Returns:
+            the number of patches in the dataset
+        """
+        return len(self.fp_patches)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """It loads the netcdf file for the given index and returns the sample as a dict.
+
+        Args:
+            index: index of the sample to return
+
+        Returns:
+            dict: a dictionary containing the sample with the following:
+
+                * the Sentinel-2 image (selected bands)
+                * the glacier mask (binary mask with all the glaciers in the current patch)
+                * the debris mask
+                * the cloud and shadow mask
+                * the additional features (DEM, derived features, etc.) if required
+        """
+        xr = lazy_import('xarray')
```

Comment:
For other datasets, we added a `lazy_import` to the `__init__` just so that the dataset would fail more quickly.